examples/surface_multi_texture.py

```diff
@@ -0,0 +1,152 @@
+If the textures don't overlap, you can display them on separate meshes as shown
+in this demo.
+
+If the textures do overlap, you may instead be able to combine the textures as
+images. This relies on textures having the same texture coordinates, and may
+require resizing the textures to match each other.
+
+.. tags:: visualization-nD
+"""
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pooch
+from vispy.io import imread, read_mesh
+
+import napari
+
+###############################################################################
+# Download the model
+# ------------------
+download = pooch.DOIDownloader(progressbar=True)
+doi = "10.6084/m9.figshare.22348645.v1"
+tmp_dir = pooch.os_cache("napari-surface-texture-example")
+os.makedirs(tmp_dir, exist_ok=True)
+data_files = {
+    "mesh": "PocilloporaDamicornisSkin.obj",
+    # "materials": "PocilloporaVerrugosaSkinCrop.mtl",  # not yet supported
+    "Texture_0": "PocilloporaDamicornisSkin_Texture_0.jpg",
+    "GeneratedMat2": "PocilloporaDamicornisSkin_GeneratedMat2.png",
+}
+print(f"downloading data into {tmp_dir}")
+for file_name in data_files.values():
+    if not (tmp_dir / file_name).exists():
+        print(f"downloading {file_name}")
+        download(
+            f"doi:{doi}/{file_name}",
+            output_file=tmp_dir / file_name,
+            pooch=None,
+        )
+    else:
+        print(f"using cached {tmp_dir / file_name}")
+
+###############################################################################
+# Load the model
+# --------------
+# Next, the model data fron the wavefront (.obj) file. Currently napari/vispy
+# do not support reading material properties (wavefront .mtl files) nor
+# separate texture and vertex indices (i.e. repeated vertices). Normal vectors
+# read from the file are also ignored and re-calculated from the faces.
+vertices, faces, _normals, texcoords = read_mesh(tmp_dir / data_files["mesh"])
+
+###############################################################################
+# Load the textures
+# -----------------
+# This model comes with two textures: `Texture_0` is generated from
+# photogrammetry of the actual object, and `GeneratedMat2` is a generated
+# material to fill in  parts of the model lacking photographic texture. The
+# texture images need to be flipped because of `how OpenGL expects the texture
+# data to be ordered in memory
+# <https://registry.khronos.org/OpenGL-Refpages/gl4/html/glTexImage2D.xhtml>`_:
+#
+#   The first element corresponds to the lower left corner of the texture
+#   image. Subsequent elements progress left-to-right through the remaining
+#   texels in the lowest row of the texture image, and then in successively
+#   higher rows of the texture image. The final element corresponds to the
+#   upper right corner of the texture image.
+photo_texture = np.flipud(imread(tmp_dir / data_files["Texture_0"]))
+generated_texture = np.flipud(imread(tmp_dir / data_files["GeneratedMat2"]))
```

Comment:
Yeah, reading this, it sounds like textures will *always* need to be flipped when loaded in as NumPy arrays, so we should bake that into how we process them internally...? It's not an easy choice, but I think users will expect:

1. when they read them into NumPy, they'll look how they would look in any image viewer. (So, I think basically imageio.v3.imread are actually doing the flipping for us somewhere in the first place?)
2. they should be able to pass them in as textures without flipping them manually.

If we accept those two axioms, then I think the conclusion is that we run flipud internally, rather than require users to do it?

(If we go this route, then please put this comment wherever you do this flipud in the napari code base — I think it should be right before passing it to VisPy — so that future code readers know what that's doing. 😃

Reply:
I think this will be less-surprising behavior for napari end-users.